client/trino-cli/src/test/java/io/trino/cli/TestClientRedirect.java

```diff
@@ -0,0 +1,191 @@
+import okhttp3.logging.HttpLoggingInterceptor;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.sql.SQLException;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.Properties;
+
+import static com.google.common.io.ByteStreams.nullOutputStream;
+import static com.google.common.net.HttpHeaders.AUTHORIZATION;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static com.google.common.net.HttpHeaders.LOCATION;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static io.trino.cli.ClientOptions.OutputFormat.CSV;
+import static io.trino.cli.TerminalUtils.getTerminal;
+import static io.trino.cli.TestQueryRunner.createClientSession;
+import static io.trino.client.ClientStandardTypes.BIGINT;
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_METHOD;
+
+@TestInstance(PER_METHOD)
+public class TestClientRedirect
+{
+    private static final JsonCodec<QueryResults> QUERY_RESULTS_CODEC = jsonCodec(QueryResults.class);
+    private static final String DUMMY_ACCESS_TOKEN = Base64.getEncoder().encodeToString("jwt-signing-key".getBytes(US_ASCII));
+    private MockWebServer server;
+
+    @BeforeEach
+    public void setup()
+            throws IOException
+    {
+        server = new MockWebServer();
+        server.start();
+    }
+
+    @AfterEach
+    public void teardown()
+            throws IOException
+    {
+        server.close();
+        server = null;
+    }
+
+    @Test
+    public void testDisableClientRedirectsDefault()
+            throws Exception
+    {
+        server.enqueue(new MockResponse()
+                .setResponseCode(307)
+                .addHeader(LOCATION, server.url("redirect/v1/statement"))
+                .addHeader(AUTHORIZATION, DUMMY_ACCESS_TOKEN));
+        server.enqueue(new MockResponse()
+                .addHeader(CONTENT_TYPE, "application/json")
+                .setBody(createResults(server)));
+
+        QueryRunner queryRunner = createQueryRunner(createTrinoUri(server, Optional.empty()), createClientSession(server));
+
+        try (Query query = queryRunner.startQuery("SELECT 1")) {
+            query.renderOutput(getTerminal(), nullPrintStream(), nullPrintStream(), CSV, Optional.of(""), false);
+        }
+
+        //verify the redirect request has the same Authorization header
+        RecordedRequest redirectedRequest = server.takeRequest();
+        assertThat(redirectedRequest.getHeader(AUTHORIZATION)).isEqualTo("Bearer " + DUMMY_ACCESS_TOKEN);
```

Comment:
This tests the first request (to proxy), which always contain the AUTHORIZATION header.
We want the second request (to Trino) to contain the AUTHORIZATION header.